typed-racket-lib/typed-racket/types/printer.rkt

```diff
@@ -478,16 +476,15 @@
 ;; case-lambda->sexp : Type -> S-expression
 ;; Convert a case-> type to an s-expression
 (define (case-lambda->sexp type)
-  (match type
```

Comment:
These two matches could be combined which would be better.

Reply:
I'm not sure I agree, particularly since the third branch of the second match wants to refer to the `arrows` variable anyway. This is how I'd write it:

```scheme
(match-define (Fun: arrows) type)
(match arrows
  [(list) '(case->)]
  [(list a) (arr->sexp a)]
  [(list a b ...)
   (define cover (cover-case-lambda arrows))
   (if (> (length cover) 1)
       `(case-> ,@cover)
       (car cover))])
```

Reply:
I guess I think I prefer:

```
(match type
  [(Fun: (list)) '(case->)]
  [(Fun: (list a)) (arr->sexp a)]
  [(Fun: arrows)
   (define cover (cover-case-lambda arrows))
   (if (> (length cover) 1)
       `(case-> ,@cover)
       (car cover))])
```

but that might be hard to generate automatically

Reply:
That looks quite reasonable. However, this all seems way too subjective for Resyntax to tackle.